perceval/backends/core/git.py

```diff
@@ -764,19 +763,26 @@ def _handle_trailer(self, line):
     def _handle_action_data(self, data):
         modes = self.__parse_data_list(data['modes'])
         indexes = self.__parse_data_list(data['indexes'])
-        filename = data['file']
+        filename = data['newfile'] if data['newfile'] else data['file']
 
         if filename not in self.commit_files:
             self.commit_files[filename] = {}
 
         self.commit_files[filename]['modes'] = modes
         self.commit_files[filename]['indexes'] = indexes
         self.commit_files[filename]['action'] = data['action']
-        self.commit_files[filename]['file'] = filename
+        self.commit_files[filename]['file'] = data['file']
         self.commit_files[filename]['newfile'] = data['newfile']
 
     def _handle_stats_data(self, data):
-        filename = self.__get_old_filepath(data['file'])
+        if ' => ' in data['file']:
+            filename = self.__get_new_filepath(data['file'])
+            if not self.merge_commit and self.commit_files and filename not in self.commit_files:
```

Comment:
Why shouldn't do this for merge commits?  They can also have actions where files are copied, renamed, etc.

Reply:
The problem with merge commits is that they behave the opposite way. They show the `numstat` output first and then the `diff-tree`. I've seen some cases where `numstat` reports renames that don't appear in the `diff-tree` output for merge commits. That’s why, in the case of merges, I avoid trying to guess the output.

Reply:
What do you mean with the `diff-tree`? I think we don't get anything like that.

For merge commits, the idea is to find those files that already change on that commit. The merge commit will list all the stats of the commits that are being merged. We parse that but the only files we need to be correct are the ones that change on that commit.